enterprise/frontend/src/metabase-enterprise/redux.ts

```diff
@@ -0,0 +1,10 @@
+import type { TypedUseSelectorHook } from "react-redux";
+import { createSelectorHook } from "react-redux";
+
+import { MetabaseReduxContext } from "metabase/lib/redux";
+
+// TODO: use the real type after we figure out what it is
```

Comment:
It seems like we pass the custom state every time, we have a few `EnterpriseState` types in the code. I'm not sure if this is because all the ee slices are optional because of the plugins/feature tokens.
At the moment, the places where this hooks is used are passing a custom type for the state.

Reply:
I actually would love to find a real type so I created https://github.com/metabase/metabase/issues/51400